libs/chonky/src/lib.rs

```diff
@@ -1,26 +1,270 @@
+
+        for page in pdf.pages().iter() {
+            // Render the entire page to a bitmap
+            page.render_into_bitmap_with_config(&mut bitmap, &config)
+                .change_context(ChonkyError::Pdfium)?;
+
+            // Convert PdfBitmap to DynamicImage
+            let dynamic_image = as_image(&bitmap)?;
+            images.push(dynamic_image);
+        }
+
+        Ok(images)
+    }
+
+    /// Takes in a config with desired resolution of image and a pdf and creates
+    /// an empty bitmap that can be used by the entire pdf
+    ///
+    /// # Errors
+    ///
+    /// Return an [`PdfiumError::CoordinateConversionFunctionIndicatedError`] if there was a problem
+    /// with converting the pdfs dimensions to pixels
+    fn create_empty_bitmap<'a>(
+        pdf: &'a PdfDocument,
+        config: &'a PdfRenderConfig,
+    ) -> Result<PdfBitmap<'a>, Report<ChonkyError>> {
+        // read the first page to get page dimesnions
+        let page_dimensions = pdf
+            .pages()
+            .page_sizes()
+            .change_context(ChonkyError::Pdfium)?[0];
+
+        let page = pdf.pages().get(0).change_context(ChonkyError::Pdfium)?;
+
+        //converts the boundings boxes to pixels that can be used for creating bitmap with proper
+        // for correct bitmap dimensions, height point must be set to 0 to get max pixel height
+        let page_dimensions = pdf
+            .pages()
+            .get(0)
+            .change_context(ChonkyError::Pdfium)?
+            .points_to_pixels(page_dimensions.width(), PdfPoints::new(0.0), config)
+            .change_context(ChonkyError::Pdfium)?;
+
+        // some pdfs have dimensions that are larger than the actual "content" resulting in negative
+        // dimensions we aim to get the "net" dimensions by subtracting the 0.0 point pixel
+        // conversion from the height and width
+        let base_page_dimensions = page
+            .points_to_pixels(PdfPoints::new(0.0), page.height(), config)
+            .change_context(ChonkyError::Pdfium)?;
+
+        let page_dimensions = (
+            page_dimensions.0 - base_page_dimensions.0,
+            page_dimensions.1 - base_page_dimensions.1,
+        );
+
+        //create an empty bitmap that follows dimensions of pdf
+        // to prevent repeated memory allocations (we assume all pdfs are same dimension)
+        let bitmap = PdfBitmap::empty(
+            page_dimensions.0,
+            page_dimensions.1,
+            PdfBitmapFormat::BGRA,
+            pdf.bindings(),
+        )
+        .change_context(ChonkyError::Pdfium)?;
+
+        Ok(bitmap)
+    }
+
+    ///A vendored function from pdfium-render's `as_image` function that returns a result instead
+    /// of panicking
+    ///
+    /// Errors#
+    ///
+    /// [`PdfiumError::ImageError`] when the image had an error being processed
+    fn as_image(bitmap: &PdfBitmap) -> Result<DynamicImage, ChonkyError> {
```

Comment:
To capture lines on errors it probably make sense to return a `Report` here:
```suggestion
    fn as_image(bitmap: &PdfBitmap) -> Result<DynamicImage, Report<ChonkyError>> {
```